Add check for outdated packages, and show on the help screen.

McInkay opened the pull request.

xPaw commented on src/packageManager.js:
```diff
@@ -0,0 +1,35 @@
+"use strict";
```
We have `src/plugins/packages/index.js`, which is probably better suited to be there

McInkay replied:
Yeah, good shout. I'll do that when I fix the conflicts for this.

McInkay commented on src/packageManager.js:
```diff
@@ -0,0 +1,35 @@
+const log = require("./log");
+const Utils = require("./command-line/utils");
+
+const outdated = () => {
```
Add caching to this, similar to `changelog.fetch`

McInkay commented:
Ok, I have moved the code into the `packages/index.js` and I have added a cache for it.

I do want to add a way for `thelounge upgrade` to clear that cache, but we currently don't have a way to talk between lounge processes, and that's probably bigger than this PR.

xPaw commented on src/plugins/packages/index.js:
```diff
@@ -86,3 +95,44 @@ function loadPackages(clientManager) {
+	// If we get an error from calling outdated and the code isn't 0, then there are no outdated packages
+	await Utils.executeYarnCommand(...argsList)
+		.then(() => updateOutdated(false))
+		.catch((code) => updateOutdated(code === 0 ? false : true));
```
```suggestion
		.catch((code) => updateOutdated(code !== 0));
```

McInkay replied:
Oh duh, rookie error. I was clearly half asleep.

xPaw commented on src/plugins/packages/index.js:
```diff
@@ -86,3 +95,44 @@ function loadPackages(clientManager) {
+	// Check if the configuration file exists
+	if (!fs.existsSync(packagesConfig)) {
+		log.warn("There are no packages installed.");
+		return;
```
Probably should be

```suggestion
		return false;
```

xPaw commented on src/plugins/packages/index.js:
```diff
@@ -86,3 +95,44 @@ function loadPackages(clientManager) {
 }
+
+async function outdated() {
+	if (cache.outdated !== undefined) {
```
```suggestion
	if (typeof cache.outdated !== "undefined") {
```

McInkay replied:
I'm not sure this is unnecessary, because we specifically specify it is undefined and then delete it, so when would this be different?

xPaw commented on src/command-line/outdated.js:
```diff
@@ -0,0 +1,20 @@
+
+const program = require("commander");
+const Utils = require("./utils");
+const packageManager = require("../packageManager");
```
Error: Cannot find module '../packageManager'

McInkay replied:
Yeah, fixed this.

xPaw commented:
I think the `outdated` command should print which packages are out of date.

McInkay commented:
> I think the outdated command should print which packages are out of date.

Can we add that in another PR? It's much bigger than what I'm doing here, and is much more complicated. This is useful in and of itself, and we can add that later.

McInkay commented:
I think this is ready? Unless there's something I missed.
